task1/views.py

```diff
@@ -0,0 +1,141 @@
+                    first_name = form.cleaned_data.get("first_name")
+                    last_name = form.cleaned_data.get("last_name")
+                    username = form.cleaned_data.get("username")
+                    password = form.cleaned_data.get("password")
+                    confirm_password = form.cleaned_data.get(
+                        "confirm_password",
+                    )
+
+                    if password != confirm_password:
+                        messages.error(request, "Password Does not match")
+                        return redirect("signup")
+                    user = User.objects.create_user(
+                        first_name=first_name,
+                        last_name=last_name,
+                        username=username,
+                        password=password,
+                    )
+                    user.save()
+                    return redirect("login")
+                else:
+                    messages.error(request, "Invalid Entry")
+                    return redirect("signup")
+            except PermissionDenied as e:
+                return HttpResponseNotFound(f"Not found {e}")
+            except Exception as e:
+                return HttpResponseServerError(f"An error occurred: {e}")
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+
+class LoginView(View):
+    template_name = "task1/login.html"
+    form_class = LoginForm
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request, *args, **kwargs):
+        if request.method == "POST":
+            form = self.form_class(request.POST)
+            try:
+                if form.is_valid():
+                    username = form.cleaned_data.get("username")
+                    password = form.cleaned_data.get("password")
+
+                    user = authenticate(username=username, password=password)
+
+                    if user is not None:
+                        login(request, user)
+                        print(f"{user} logged in")
+                        return redirect("home")
+                    else:
+                        messages.error(request, "Invalid username or password")
+                        return redirect("login")
+                else:
+                    messages.error(request, "Invalid form submission")
+                    return render(request, "task1/login.html", {"form": form})
+            except PermissionDenied as e:
+                return HttpResponseNotFound(f"Not found {e}")
+            except Exception as e:
+                return HttpResponseServerError(f"An error occurred: {e}")
+        form = self.form_class()
+        return render(request, self.template_name, {"form": form})
+
+
+class LogoutView(View):
+    def get(self, request, *args, **kwargs):
+        print(f"{request.user.username} logged out")
```

Comment:
you can use logger instead of print here

Reply:
sure